node.go

```diff
@@ -71,7 +75,7 @@ func (conn *connection) Error(code C.int) error {
 // another thread). There is a mutable version of Node emitted by Node iterators (FOR loops over Node), which
 // may not be shared with other threads except by first taking an immutable Node.Copy() of it.
 type node struct {
-	conn    connection
+	conn    *connection
```

Comment:
This is the main problem, I think, and why the connection is getting freed before you expect. Your node was making a shallow copy of the connection struct. But what you want it to do is reference the original one. That fixes the KeepAlive / GC issue you were having.